L1Trigger/TrackFindingTracklet/test/AnalyzerTracklet.cc

```diff
@@ -0,0 +1,247 @@
+    hisChannel_ = dir.make<TH1F>("His Channel Occupancy", ";", maxOcc, -.5, maxOcc - .5);
+    profChannel_ = dir.make<TProfile>("Prof Channel Occupancy", ";", numChannels, -.5, numChannels - .5);
+    // Efficiencies
+    hisEffTotal_ = dir.make<TH1F>("HisTPEtaTotal", ";", 128, -2.5, 2.5);
+    hisEff_ = dir.make<TH1F>("HisTPEta", ";", 128, -2.5, 2.5);
+    eff_ = dir.make<TEfficiency>("EffEta", ";", 128, -2.5, 2.5);
+  }
+
+  void AnalyzerTracklet::analyze(const Event& iEvent, const EventSetup& iSetup) {
+    auto fill = [this](const TPPtr& tpPtr, TH1F* his) { his->Fill(tpPtr->eta()); };
+    // read in tracklet products
+    Handle<TTTracks> handle;
+    iEvent.getByToken<TTTracks>(edGetToken_, handle);
+    // read in MCTruth
+    const StubAssociation* selection = nullptr;
+    const StubAssociation* reconstructable = nullptr;
+    if (useMCTruth_) {
+      Handle<StubAssociation> handleSelection;
+      iEvent.getByToken<StubAssociation>(edGetTokenSelection_, handleSelection);
+      selection = handleSelection.product();
+      prof_->Fill(9, selection->numTPs());
+      Handle<StubAssociation> handleReconstructable;
+      iEvent.getByToken<StubAssociation>(edGetTokenReconstructable_, handleReconstructable);
+      reconstructable = handleReconstructable.product();
+      for (const auto& p : selection->getTrackingParticleToTTStubsMap())
+        fill(p.first, hisEffTotal_);
+    }
+    //
+    const TTTracks& ttTracks = *handle.product();
+    vector<TTTrackRefs> ttTrackRefsRegions(setup_->numRegions());
+    vector<int> nTTTracksRegions(setup_->numRegions(), 0);
+    for (const TTTrack<Ref_Phase2TrackerDigi_>& ttTrack : ttTracks)
+      nTTTracksRegions[ttTrack.phiSector()]++;
+    for (int region = 0; region < setup_->numRegions(); region++)
+      ttTrackRefsRegions[region].reserve(nTTTracksRegions[region]);
+    int i(0);
+    for (const TTTrack<Ref_Phase2TrackerDigi_>& ttTrack : ttTracks)
+      ttTrackRefsRegions[ttTrack.phiSector()].emplace_back(TTTrackRef(handle, i++));
+    for (int region = 0; region < setup_->numRegions(); region++) {
+      const TTTrackRefs& ttTrackRefs = ttTrackRefsRegions[region];
+      const int nStubs = accumulate(ttTrackRefs.begin(), ttTrackRefs.end(), 0, [](int& sum, const TTTrackRef& ttTrackRef){ return sum += ttTrackRef->getStubRefs().size(); });
+      const int nTracks = ttTrackRefs.size();
+      hisChannel_->Fill(nTracks);
+      profChannel_->Fill(region, nTracks);
+      prof_->Fill(1, nStubs);
+      prof_->Fill(2, nTracks);
+      prof_->Fill(3, 0);
+    }
+    // analyze tracklet products and associate found tracks with reconstrucable TrackingParticles
+    set<TPPtr> tpPtrs;
+    set<TPPtr> tpPtrsSelection;
+    int allMatched(0);
+    vector<vector<TTStubRef>> tracks;
+    tracks.reserve(ttTracks.size());
+    transform(ttTracks.begin(), ttTracks.end(), back_inserter(tracks), [](const TTTrack<Ref_Phase2TrackerDigi_>& ttTrack){ return ttTrack.getStubRefs(); });
```

Comment:
Add comment "Convert vector of tracks to vector of vector of associated stubs"

Reply:
done